amcl/src/amcl_node.cpp

```diff
@@ -477,6 +499,43 @@ AmclNode::AmclNode() :
   laser_check_interval_ = ros::Duration(15.0);
   check_laser_timer_ = nh_.createTimer(laser_check_interval_,
                                        boost::bind(&AmclNode::checkLaserReceived, this, _1));
+
+  
+
+  //start action server
+  set_inital_pose_action_.start();
+
+  brainstem_driver_pose_pub_ = nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>(srs::ChuckTopics::internal::ODOMETRY_INITIAL_POSE, 2, true);
+  publishAmclReadySignal(true);
+}
+
+void AmclNode::publishAmclReadySignal(bool signal)
+{
+  // publish ready signal
+  ready_pub_.publish(signal);
+}
+
+void AmclNode::executeInitialPoseCB(const move_base_msgs::SetInitialPoseGoalConstPtr &goal)
+{
+  ROS_INFO("Executing, inital pose server action");
+  if(use_emulator_)
+  {
+    brainstem_driver_pose_pub_.publish(goal->initialPose);
```

Comment:
so if emulation mode is enabled, the brainstem driver looks at a different topic. Is there some diagram to show what this all is doing and how it changed.

Reply:
if emulation is on the odomtery doesn't come from brainstem emulator so we don't need to publish on this topic to republish new tfs. I can document it in wiki page for this feature